components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -971,6 +975,25 @@ void BraveWalletService::OnGetNftsForCompressedMigration(
   profile_prefs_->SetBoolean(kBraveWalletIsCompressedNftMigrated, true);
 }
 
+void BraveWalletService::MaybeMigrateSPLTokenProgram() {
+  if (profile_prefs_->GetBoolean(kBraveWalletIsSPLTokenProgramMigrated)) {
+    return;
+  }
+
+  // Get all solana SPL NFTs that are marked incorrectly as unsupported
+  // and reset their spl_token_program to unknown.
+  for (auto& item : ::brave_wallet::GetAllUserAssets(profile_prefs_)) {
+    if (item->coin == mojom::CoinType::SOL && item->is_nft &&
+        item->spl_token_program == mojom::SPLTokenProgram::kUnsupported &&
+        IsSPLToken(item)) {
```

Comment:
IsSPLToken(item) covers item->coin == mojom::CoinType::SOL so we can remove item->coin == mojom::CoinType::SOL here.